### PR TITLE
[16.0] [MIG] Migrated label

### PR DESCRIPTION
**Pull request by dhaval-serpentcs**

### CODE REVIEW

File: label/LICENSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,16 @@
+    OpenERP, Open Source Management Solution
+    Copyright (C) 2011-Today Serpent Consulting Services Pvt. Ltd.
```
Comment (Trivedi-Vacha-SerpentCS) [inline label/LICENSE/LICENSE]:
```suggestion
    Copyright (C) 2023-Today Serpent Consulting Services Pvt. Ltd.
```

File: label/README.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+==================
+Label
+==================
```
Comment (Trivedi-Vacha-SerpentCS) [inline label/README.rst]: keep == till last l of "Label"

File: label/models/label_print.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                }
+            )
+        self.write({"ref_ir_act_report": vals.get(
+            "ref_ir_act_report", False).id})
+        return True
+
+    def unlink_action(self):
+        for template in self:
+            if template.ref_ir_act_report.id:
+                template.ref_ir_act_report.unlink()
```
Comment (Trivedi-Vacha-SerpentCS) [inline label/models/label_print.py]: use inline for loop

File: label/report/dynamic_label.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+from odoo import _, api, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval as eval
+
+
+class ReportDynamicLabel(models.AbstractModel):
+    _name = "report.label.report_label"
+    _description = "Print Label Report Class"
+
+    def get_data(self, row, columns, ids, model, number_of_copy):
```
Comment (Trivedi-Vacha-SerpentCS) [inline label/report/dynamic_label.py:15]: Good if we can optimise this method

File: label/wizard/label_print_wizard.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    def print_report(self):
+        if self._context is None:
+            self._context = {}
+        if not self._context.get("label_print") or not \
+                self._context.get("active_ids"):
+            return False
+        total_record = len(self._context.get("active_ids", []))
+        datas = {}
+        for data in self.browse(self.ids):
```
Comment (Trivedi-Vacha-SerpentCS) [inline label/wizard/label_print_wizard.py]: what is the reason browse self? can't we do for data in self:

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by dhaval-serpentcs** [comment]
i will improve feedback points and send pr from new branch.

**Pull request closed without merging**